extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -77,40 +85,129 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             deviceClient = new TradfriCoapClient(uri);
         } catch (URISyntaxException e) {
-            logger.debug("Illegal gateway URI `{}`: {}", gatewayURI, e.getMessage());
+            logger.error("Illegal gateway URI '{}': {}", gatewayURI, e.getMessage());
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
             return;
         }
 
         DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
-        builder.setPskStore(new StaticPskStore("", configuration.code.getBytes()));
+        builder.setPskStore(new StaticPskStore(configuration.identity, configuration.preSharedKey.getBytes()));
         dtlsConnector = new DTLSConnector(builder.build());
         endPoint = new TradfriCoapEndpoint(dtlsConnector, NetworkConfig.getStandard());
         deviceClient.setEndpoint(endPoint);
-        updateStatus(ThingStatus.UNKNOWN);
 
         // schedule a new scan every minute
-        scanJob = scheduler.scheduleWithFixedDelay(() -> {
-            startScan();
-        }, 0, 1, TimeUnit.MINUTES);
+        scanJob = scheduler.scheduleWithFixedDelay(this::startScan, 0, 1, TimeUnit.MINUTES);
+    }
+
+    protected void obtainIdentityAndPreSharedKey() {
+        TradfriGatewayConfig configuration = getConfigAs(TradfriGatewayConfig.class);
+
+        String identity = UUID.randomUUID().toString().replace("-", "");
+        String preSharedKey = null;
+
+        CoapResponse gatewayResponse;
+        String authUrl = null;
+        String responseText = null;
+        try {
+            DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
+            builder.setPskStore(new StaticPskStore("Client_identity", configuration.code.getBytes()));
+
+            DTLSConnector dtlsConnector = new DTLSConnector(builder.build());
+            CoapEndpoint authEndpoint = new CoapEndpoint(dtlsConnector, NetworkConfig.getStandard());
+            authUrl = "coaps://" + configuration.host + ":" + configuration.port + "/15011/9063";
+
+            CoapClient deviceClient = new CoapClient(new URI(authUrl));
+            deviceClient.setEndpoint(authEndpoint);
+
+            JsonObject json = new JsonObject();
+            json.addProperty(CLIENT_IDENTITY_PROPOSED, identity);
+
+            gatewayResponse = deviceClient.post(json.toString(), 0);
+
+            authEndpoint.destroy();
+            deviceClient.shutdown();
+
+            if (gatewayResponse.isSuccess()) {
+                responseText = gatewayResponse.getResponseText();
+                json = new JsonParser().parse(responseText).getAsJsonObject();
+                preSharedKey = json.get(NEW_PSK_BY_GW).getAsString();
+
+                if (isNullOrEmpty(preSharedKey)) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Pre-shared key was not obtain successfully");
```

Comment:
Can we ever get here? Imho it is a bug if we get here, so you should log an ERROR here - and it is not a configuration error as the user cannot do anything about it, can he?

Reply:
I don't think we can do anything here - will add an error log.
and what about the Thing Status - should I set it to ```COMMUNICATION_ERROR``` or ```CONFIGURATION_ERROR``` is OK?

Reply:
`COMMUNICATION_ERROR` as what we received does not make sense to us.